### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by kderme**
This will become relevant after https://github.com/input-output-hk/cardano-ledger-specs/pull/1536

### CODE REVIEW

File: ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano.hs [hunk 1]
```diff
@@ -140,13 +142,15 @@ data Protocol (m :: * -> *) blk p where
   ProtocolRealTPraos
     :: ShelleyGenesis TPraosStandardCrypto
     -> ProtVer
+    -> Natural
```
Comment (mrBliss) [inline ouroboros-consensus-cardano/src/Ouroboros/Consensus/Cardano.hs:145]:
```suggestion
    -> Natural  -- ^ Max major protocol version
```
Same for L153.

Better would be to introduce a newtype `MaxMajorPV`.

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs [hunk 2]
```diff
@@ -84,41 +82,14 @@ data TPraosLeaderCredentials c = TPraosLeaderCredentials {
...
-
-    --TODO: move this to the hash API module
-    castHash :: Crypto.Hash (HASH c) a -> Crypto.Hash (HASH c) b
-    castHash (Crypto.UnsafeHash h) = Crypto.UnsafeHash h
-
 protocolInfoShelley
   :: forall m c. (MonadRandom m, TPraosCrypto c)
-  => ShelleyGenesis c
+  => SL.ShelleyGenesis c
+  -> Natural
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:88]: Same thing here

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs [hunk 3]
```diff
@@ -135,27 +106,27 @@ protocolInfoShelley genesis protVer mbCredentials =
...
-      , tpraosMaxMajorPV        = sgMaxMajorPV        genesis
-      , tpraosMaxLovelaceSupply = sgMaxLovelaceSupply genesis
-      , tpraosNetworkId         = sgNetworkId         genesis
+        tpraosSlotsPerKESPeriod = SL.sgSlotsPerKESPeriod genesis
+      , tpraosLeaderF           = SL.sgActiveSlotCoeff   genesis
+      , tpraosSecurityParam     = SecurityParam $ SL.sgSecurityParam genesis
+      , tpraosMaxKESEvo         = SL.sgMaxKESEvolutions  genesis
+      , tpraosQuorum            = SL.sgUpdateQuorum      genesis
+      , tpraosMaxLovelaceSupply = SL.sgMaxLovelaceSupply genesis
+      , tpraosNetworkId         = SL.sgNetworkId         genesis
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:129]: Realign on `genesis`. Better yet, add:
```haskell
securityParam :: SecurityParam
securityParam = SecurityParam $ SL.sgSecurityParam genesis
```
(at the same level as `tpraosParams` is fine.)

**Comment by mrBliss** [comment]
Maybe you can bump the dependencies again, some things probably have changed in the meantime.

**Comment by kderme** [comment]
bors merge

**Pull request merged**